build-aux/genversion.go

```diff
@@ -39,9 +39,16 @@ func Main() error {
 	}
 	gitDescVer.Patch++
 
-	_, err = fmt.Printf("v%s-%d\n", gitDescVer, time.Now().Unix())
-	if err != nil {
-		return err
+	// If an additional arg has been used, we include it in the tag
+	if len(os.Args) >= 2 {
+		t := time.Now()
+		date := fmt.Sprintf("%d.%d.%d", t.Year(), t.Month(), t.Day())
+		_, err = fmt.Printf("v%d.%d.%d-%s-%s-%s\n", gitDescVer.Major, gitDescVer.Minor, gitDescVer.Patch, os.Args[1], date, gitDescVer.Pre[0].String())
```

Comment:
`gitDescVer.Pre[0]` is the number of commits since the previous tag, and `gitDescVer.Pre[1]` is the short hash of the commit (with a `g` prefix), right?

I'd think it would be sufficiently useful to include the shorthash, so it's easier to figure out which commit a nightly version was built from.